cpp/src/table/row_operators.cu

```diff
@@ -264,18 +282,23 @@ using column_checker_fn_t = std::function<void(column_view const&)>;
  */
 void check_lex_compatibility(table_view const& input)
 {
-  // Basically check if there's any LIST hiding anywhere in the table
+  // Basically check if there's any LIST of STRUCT or STRUCT of LIST hiding anywhere in the table
   column_checker_fn_t check_column = [&](column_view const& c) {
-    CUDF_EXPECTS(c.type().id() != type_id::LIST,
-                 "Cannot lexicographic compare a table with a LIST column");
+    if (c.type().id() == type_id::LIST) {
+      auto const& list_col = lists_column_view(c);
+      CUDF_EXPECTS(list_col.child().type().id() != type_id::STRUCT,
+                   "Cannot lexicographic compare a table with a LIST of STRUCT column");
+      check_column(list_col.child());
+    } else if (c.type().id() == type_id::STRUCT) {
+      for (auto child = c.child_begin(); child < c.child_end(); ++child) {
+        check_column(*child);
+      }
+    }
     if (not is_nested(c.type())) {
       CUDF_EXPECTS(is_relationally_comparable(c.type()),
                    "Cannot lexicographic compare a table with a column of type " +
                      jit::get_type_name(c.type()));
```

Comment:
I'd like to eliminate the dependence on the `cudf::jit` namespace here, perhaps in a near-term follow-up PR to this one. There's even a three year old TODO to remove that function entirely.

https://github.com/rapidsai/cudf/blob/bc7109e9a73a71a0de2da225c8da03ad27fea73f/cpp/src/jit/type.cpp#L73

Reply:
Let's do it in a follow-up. I don't want to hold up this PR any longer for unrelated changes.

Reply:
(Is this follow up work being tracked? I usually don’t click “Resolve Conversation” if later work is needed unless an issue has been filed.)

Reply:
#11673